test code for euclidean_distance

Seungbongjung opened the pull request:
test_distance.jl 확인 부탁드립니다!

Seungbongjung committed:
test code for euclidean_distance

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+using Test, CuArrays, DistStat, LinearAlgebra
+A=CuArray; type=[Float64, Float32]
```
An assumption for the test will be that GPU is not always available. See https://github.com/kose-y/DistStat.jl/blob/master/test/test_diag.jl#L5

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+for T in type
+    test1=[1.0 1.0 1.0; -1.0 0.0 2.0; 0.0 1.0 3.0]
+    test2=[1.0 2.0 3.0; -1.0 0.0 1.0; 2.0 0.0 1.0]
+    test1_cu=cu(test1); test1_dist=distribute(test1_cu)
```
Change any `cu(x)` into `convert(ArrayType{T}, x)`. The code should run even when GPU is not available.

kose-y replied:
also change the names of variables ending with `_cu`.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+            answer1=sqrt(sum((test1[:,i]-test1[:,j]).^2))
+            answer2=sqrt(sum((test1[;,i]-test2[:,j]).^2))
+        end
+    end
```
Aren't `test1` and `test2` 3 x 3?

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+
+    @test isapprox(result1[diagind(result1)],zero1)
+    @test (result1 .>= 0) == criteria1
+    @test (result2 .>= 0) == criteria1
```
change to e.g. `@test all(result1 .>= 0)`.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+
+    zero1=[0.0, 0.0, 0.0]
+    criteria1=Bool[1 1 1; 1 1 1; 1 1 1]
+    criteria1=A{Bool}(criteria1)
```
no need.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+    test1_cu=cu(test1); test1_dist=distribute(test1_cu)
+    test2_cu=cu(test2); test2_dist=distribute(test2_cu)
+    answer1=Array{T}(undef,3,3); answer2=Array{T}(undef,3,3)
+    r1=A{Float32}(undef, 3, 3); r2=A{Float32}(undef,3,3)
```
Why are they fixed to `Float32`?

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+
+    criteria2=Bool[1 1 1 1 1 1 1; 1 1 1 1 1 1 1; 1 1 1 1 1 1 1; 1 1 1 1 1 1 1;
+    1 1 1 1 1 1 1; 1 1 1 1 1 1 1; 1 1 1 1 1 1 1]
+    criteria2=A{Bool}(criteria2)
```
no need.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+    1 1 1 1 1 1 1; 1 1 1 1 1 1 1; 1 1 1 1 1 1 1]
+    criteria2=A{Bool}(criteria2)
+
+    @test (result.localarray .>= 0) == criteria2
```
use `all` like above.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+    cols=test_dist.partitioning[DistStat.Rank()+1][2]
+
+    r=MPIArray{Float32,2,A}(undef,3,3)
+    answer=Array{Float64}(undef,7,7)
```
types.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+    answer=Array{Float64}(undef,7,7)
+    for i in 1:7
+        for j in 1:7
+            answer[i,j]=sqrt(sum((test[:,j]-test[:,j]).^2))
```
`sqrt(sum((test[:,i]-test[:,j]).^2))`?

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+    end
+    answer=cu(answer)
+
+    @test test_cu[:,cols] == test_dist.localarray
```
Don't really need to test this here. This should be tested in some other file.

kose-y commented on test/test_distance.jl:
```diff
@@ -0,0 +1,68 @@
+
+    zero2=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+
+    @test isapprox(result.localarray[diagind(result.localarray)],zero2)
```
is it right?

kose-y reviewed (changes requested):
일단 한 번 훑어봤습니다. 테스트가 정상적으로 작동하는지 의심스러운 부분이 몇 군데 눈에 띄는데, 확인해봤나요?

Seungbongjung committed:
test code for euclidean_distance

Seungbongjung committed:
test runs well

Seungbongjung committed:
tests code for opnorm

kose-y commented on test/test_opnorm.jl:
```diff
@@ -0,0 +1,32 @@
+using DistStat, LinearAlgebra,Random,Test
+
+type=[Float64,Float32]
+A=Array
```
CuArray?

kose-y commented on test/test_opnorm.jl:
```diff
@@ -0,0 +1,32 @@
+        end
+    end
+
+    println(@test all(isapprox(opnorm(a_dist),result)))
```
println is not necessary here

kose-y commented on test/test_opnorm.jl:
```diff
@@ -0,0 +1,32 @@
+A=Array
+
+for T in type
+    a=rand(7,7)
```
I think you misunderstood how `opnorm` works. It computes the operator norm for the whole MPIMatrix, not for each partition.

Seungbongjung committed:
modified test

Seungbongjung committed:
test runs well under certain condition

When type of a is CuArray, test doesn't run well. I think solving this problem is sufficient.

Seungbongjung committed:
test runs well

except when a is CuArray

Seungbongjung committed:
modified

Seungbongjung committed:
test runs well

Seungbongjung committed:
test runs well except for Array

Still there is a problem when the type CuArray

Seungbongjung committed:
test runs well except for CuArray

Seungbongjung committed:
Update test_mul.jl

Seungbongjung committed:
new test

Seungbongjung committed:
Update test_broadcasting.jl

Seungbongjung committed:
Test runs well

Seungbongjung committed:
test runs well

Seungbongjung committed:
Create test_mpi.jl

Seungbongjung committed:
Create test_broadcasting.jl

kose-y committed:
merge master to unittest (#7)

* memory-efficient Cox

* fix pet-l1

* Update and rename LICENSE to LICENSE.md

* Update LICENSE.md

* Update LICENSE.md

* Update README.md

* Update .travis.yml

kose-y committed:
Merge branch 'master' into unittest

kose-y committed:
mpi config files (#8)

* Create travis-install-mpi.sh

* Create ompi_rootenv.patch

kose-y committed:
add Pkg (#9)

* Create travis-install-mpi.sh

* Create ompi_rootenv.patch

* add Pkg

kose-y committed:
Remove Julia 1.0 and 1.1

kose-y committed:
remove test/Project.toml (#10)

* Create travis-install-mpi.sh

* Create ompi_rootenv.patch

* add Pkg

* remove Julia 1.0 and 1.1

* Update README.md

* Delete Project.toml

kose-y committed:
simple fixes

kose-y committed:
empty doc (#11)

* Create travis-install-mpi.sh

* Create ompi_rootenv.patch

* add Pkg

* remove Julia 1.0 and 1.1

* Update README.md

* Delete Project.toml

* Create index.md

* toml for docs
